src/notification/extract.js

```diff
@@ -70,23 +71,6 @@ const asDict = (obj: JSONableInput | void): JSONableInputDict | void => {
         ...(StreamData | PmData),
     } };
     ```
-
-    Note that prior to 1.7.0-1351-g98943a8333, we only received the
-    `message_ids` field. Messages of this form are not useful.
-
-    The pair `(server, realm_id)`, added in the same commit, uniquely identifies
-    a Zulip realm, and was originally intended to permit the client to associate
-    a notification with its associated realm. Unfortunately, there is no way to
-    get either of these from a server via the API, so this would not be possible
-    until the addition of `realm_uri` in 1.8.0-2150-g5f8d193bb7...
-
-    ... which still didn't permit differentiating between multiple accounts on
-    the same realm. This was only made possible by the addition of the `user_id`
-    field, in 2.1-dev-540-g447a517e6f.
-
-    TODO(server-1.8): Simplify this comment a lot.
-    TODO(server-1.9): Simplify further.
-    TODO(server-2.1): Simplify further.
```

Comment:
In addition to the paragraphs just above, the other simplification I had in mind here is to delete the distinctions between "antiquity" and the earliest supported version on the various properties, and to simplify the paragraphs at the top.